client/express-checkout/utils/normalize.js

```diff
@@ -7,14 +7,17 @@
  * @return {Array} An array of PaymentItems
  */
 export const normalizeLineItems = ( displayItems ) => {
-	return displayItems.map( ( displayItem ) =>
-		// The amount prop is already present on the item.
-		( {
-			...displayItem,
+	return displayItems.map( ( displayItem ) => {
+		let amount = displayItem?.amount ?? displayItem?.value;
+		if ( displayItem.key === 'total_discount' ) {
+			amount = -amount;
```

Comment:
I wonder if the discount amount should come as a negative value from the server. That way, if another item that subtracts from the total is added, we won’t need to add another condition here.

Reply:
I was wondering the same thing actually, what gave me pause is introducing a "new" pattern to the line / display items that get used. It seems everywhere else discount is described with a positive int. For example in the cart totals in WC core, the total_discount is stored as a positive int. I figured this approach is less risky for now, but like you said may need alterations if there is another line item that would need to be treated as a negative.

Reply:
It makes sense, but in this case, it only applies to the object for display purposes – Stripe seem to it though. The discount will continue to be handled as it is currently. Nevertheless we can handle later if actually need it to be that way.